android-mvvm/src/main/java/com/example/android_mvvm/ReadOnlyField.java

```diff
@@ -0,0 +1,48 @@
+package com.example.android_mvvm;
+
+import android.databinding.ObservableField;
+import android.support.annotation.NonNull;
+
+import java.util.HashMap;
+
+import rx.Observable;
+import rx.Subscription;
+import rx.functions.Action1;
+
+public class ReadOnlyField<T> extends ObservableField<T> {
+    final Observable<T> source;
+    final HashMap<OnPropertyChangedCallback, Subscription> subscriptions = new HashMap<>();
+
+    public ReadOnlyField(@NonNull Observable<T> source) {
+        this.source = source.share();
+    }
+
+    /**
+     * Calling set method of ReadOnlyField can result in weird behaviour
+     * TODO: Document why set() shouldn't be used
+     */
+    @Override
+    public void set(T value) {
+        super.set(value);
+    }
+
+    @Override
+    public synchronized void addOnPropertyChangedCallback(OnPropertyChangedCallback callback) {
+        super.addOnPropertyChangedCallback(callback);
+        subscriptions.put(callback, source.subscribe(new Action1<T>() {
```

Comment:
Yeah, this can be improved.
Solution would be to use a blank `subscribe()`, and call `set(value)` using `doOnNext()`.
`this.source = source.doOnNext(x => set(x)).share()`

Any ideas for a test to enforce this?